2-bst/binary_tree.rb

```diff
@@ -0,0 +1,51 @@
+require 'bundler'
+
+Bundler.require(:default)
+
+class BinaryTree
+  attr_reader :left
+  attr_reader :right
+  attr_reader :value
+
+  def initialize(left, right, value)
+    @left = left
+    @right = right
+    @value = value.to_s
+  end
+
+  def self.create(array)
+    return nil if array.empty?
+    array.sort!.uniq!
+    mid = array[array.length / 2]
+    left_arr = array.select { |el| el < mid }
```

Comment:
to rozwiązanie jest całkiem ładne, ale zwróć uwagę, że przechodzisz przez całą tablicę dwa razy. Kosztem minimalnie bardziej skomplikowanego kodu można sporo zaoszczędzić:
```ruby
left_arr, right_arr = array.partition { |el| el < mid }
right_arr = right_arr.drop(1)
```